src/vs/workbench/contrib/extensions/browser/extensionsWorkbenchService.ts

```diff
@@ -1042,7 +1042,8 @@ export class ExtensionsWorkbenchService extends Disposable implements IExtension
 		// The check is added here because we want to fail fast instead of downloading the VSIX and then fail.
 		if (gallery.properties.engine && (!isEngineValid(gallery.properties.engine, this.productService.vscodeVersion, this.productService.date)
 			|| (gallery.properties.azDataEngine && !isEngineValid(gallery.properties.azDataEngine, this.productService.version, this.productService.date)))) {
-			return Promise.reject(new Error(nls.localize('incompatible2', "Unable to install version '{2}' of extension '{0}' as it is not compatible with Azure Data Studio '{1}'.", extension.gallery!.identifier.id, this.productService.version, gallery.version)));
+			const error = new ExtensionManagementError(nls.localize('incompatible2', "Unable to install version '{2}' of extension '{0}' as it is not compatible with Azure Data Studio '{1}'. Update to Azure Data Studio {3} to install the extension.", extension.gallery!.identifier.id, this.productService.version, gallery.version, gallery.properties.azDataEngine), INSTALL_ERROR_INCOMPATIBLE);
```

Comment:
Oh and when changing VS strings we need to add them to https://github.com/Microsoft/azuredatastudio/blob/ffdefd3b52ec5b2acb6620c0707f99bc0f32faa0/src/sql/base/common/locConstants.ts otherwise they'll get clobbered by the VS langpack strings

Reply:
Although actually - if this isn't a string that VS code has themselves we don't need to do that. So check that first and if they don't then this is fine

Reply:
I don't think this is a vscode string as 'Azure Data Studio' in it. Vscode has its own extension installation errors

Reply:
Good point 😄